asapdiscovery/simulation/scripts/MproP0009.py

```diff
@@ -39,18 +32,67 @@ def get_args():
         default = "/data/chodera/lemonsk/asap-datasets/prepped_mpro_P0009/",
         help="Output simulation directory."
     )
+    parser.add_argument(
+        "-l",
+        "--ligand",
+        type = str,
+        default = "/data/chodera/lemonsk/covid-moonshot-ml/asapdiscovery/simulation/tests/inputs/MAT-POS-f2460aef-1.sdf",
+        help = "Ligand"
+    )
+    parser.add_argument(
+        "-p",
+        "--protein",
+        type = str,
+        default = "/data/chodera/lemonsk/asap-datasets/openmm_setup_processed/prepped_receptor_0-processed.pdb",
+        help = "Protein-receptor"
+    )
     args = parser.parse_args()
     return args
 
 args = get_args()
 
+from rdkit import Chem
+rdkitmol = Chem.SDMolSupplier(args.ligand)[0]
+rdkitmolh = Chem.AddHs(rdkitmol, addCoords=True);
+# ensure the chiral centers are all defined
+Chem.AssignAtomChiralTagsFromStructure(rdkitmolh);
+from openff.toolkit.topology import Molecule
+ligand_mol = Molecule(rdkitmolh);
+
+# Initialize a SystemGenerator
+from openmm import unit, app
+from openmmforcefields.generators import SystemGenerator
+forcefield_kwargs = {'constraints': app.HBonds, 'rigidWater': True, 'removeCMMotion': False, 'hydrogenMass': 4*unit.amu }
+periodic_forcefield_kwargs = {'nonbondedMethod': app.PME}
+system_generator = SystemGenerator(
+    forcefields=['amber/ff14SB.xml', 'amber/tip3p_standard.xml'],
+    small_molecule_forcefield='openff-1.3.1',
+    molecules=[ligand_mol], cache='cache.json',
+    forcefield_kwargs=forcefield_kwargs, periodic_forcefield_kwargs=periodic_forcefield_kwargs);
+
+# Use Modeller to combine the protein and ligand into a complex
+from openmm.app import PDBFile
+from openmm.app import Modeller
+protein_pdb = PDBFile(args.protein)
+modeller = Modeller(protein_pdb.topology, protein_pdb.positions)
+# This next bit is black magic.
+# Modeller needs topology and positions. Lots of trial and error found that this is what works to get these from
+# an openforcefield Molecule object that was created from a RDKit molecule.
+# The topology part is described in the openforcefield API but the positions part grabs the first (and only)
+# conformer and passes it to Modeller. It works. Don't ask why!
+modeller.add(ligand_mol.to_topology().to_openmm(), ligand_mol.conformers[0].to_openmm())
+
 # Input Files
 
-pdb = PDBFile(args.input_pdb_path)
+from openmm.app import ForceField
+#pdb = PDBFile(args.input_pdb_path)
 forcefield = ForceField('amber14-all.xml', 'amber14/tip3pfb.xml')
```

Comment:
we don't need this line anymore since we should be using `system_generator.forcefield` as the forcefield!

Reply:
Yes, very true!